projects/HVAC/vent_control/vent.ino

```diff
@@ -0,0 +1,260 @@
+    delay(100);
+    digitalWrite(blink_pin, LOW);
+    delay(50);
+  }
+}
+
+// -----------
+
+void process_command() {
+  char buffer [100];
+  long int value = 0;
+
+  if(net.assertCommandStarts("set degrees:", buffer)) {
+    value = strtol(buffer, NULL, 10);
+    motor_rotate_degrees(value);
+    net.sendResponse("OK");
+    return;
+  }
+
+  if(net.assertCommandStarts("set percents:", buffer)) {
+    value = strtol(buffer, NULL, 10);
+    motor_rotate_percents(value);
+    net.sendResponse("OK");
+    return;
+  }
+
+  net.sendResponse("ERROR");
+}
+
+// ----------- MOTOR ROUTINES -------------
+
+void motor_step_cw() {
+  step1();
+  step2();
+  step3();
+  step4();
+  step5();
+  step6();
+  step7();
+  step8();
+}
+
+void motor_step_ccw() {
+  step8();
+  step7();
+  step6();
+  step5();
+  step4();
+  step3();
+  step2();
+  step1();
+}
+
+void step1(){
+  digitalWrite(8, HIGH);
+  digitalWrite(9, LOW);
+  digitalWrite(10, LOW);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step2(){
+  digitalWrite(8, HIGH);
+  digitalWrite(9, HIGH);
+  digitalWrite(10, LOW);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step3(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, HIGH);
+  digitalWrite(10, LOW);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step4(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, HIGH);
+  digitalWrite(10, HIGH);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step5(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, LOW);
+  digitalWrite(10, HIGH);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step6(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, LOW);
+  digitalWrite(10, HIGH);
+  digitalWrite(11, HIGH);
+  delay(2);
+}
+void step7(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, LOW);
+  digitalWrite(10, LOW);
+  digitalWrite(11, HIGH);
+  delay(2);
+}
+void step8(){
+  digitalWrite(8, HIGH);
+  digitalWrite(9, LOW);
+  digitalWrite(10, LOW);
+  digitalWrite(11, HIGH);
+  delay(2);
+}
+
+bool motor_zero() {
+  for(int i = 0; i < motor_steps_per_rotation; i++) {
+    motor_step_cw();
+    if(digitalRead(motor_sensor_pin) == HIGH) {
+      motor_position = 0;
+      return true;
+    }
+  }
+  return false;
+}
+
+void motor_rotate_to(int position) {
+  int new_position = position % motor_steps_per_rotation;
+  if(new_position > motor_position) {
+    while(motor_position != new_position) {
+      motor_step_cw();
+      motor_position++;
+    }
+  } else {
+    while(motor_position != new_position) {
+      motor_step_ccw();
+      motor_position--;
+    }
+  }
+}
+
+void motor_rotate_degrees(int degrees) {
+  int new_position = (int) (motor_steps_per_degree * (degrees % 360));
+  motor_rotate_to(new_position);
+}
+
+void motor_rotate_percents(int percents) {
+  motor_rotate_to(motor_steps_per_rotation / 100.0 * (percents % 100));
+}
+
+
+// ----------- ERROR HANDLING ROUTINES -------------
+void set_error(char* message) {
+  char buffer[100];
+  sprintf(buffer, "ERROR: %s", message);
+  net.sendResponse(buffer);
```

Comment:
You're sending this response when no one is expecting it. RS-485 is asynchronous protocol. Unless designated master asks, you shouldn't be speaking. And in the beginning, you reply with OK to each command, so by this time master would've processed your command, and might be talking to someone else, so this will introduce a collision.